src/reaction/utils/applyOneReactantReaction.js

```diff
@@ -56,23 +53,17 @@ export function applyOneReactantReaction(reactants, reactions, options) {
               products.push(product);
 
               todoNextDepth.push(() => {
-                return applyOneReactantReaction(
-                  oneReactionProducts[i][j],
-                  reactions,
-                  {
-                    ...options,
-                    currentDepth: options.currentDepth + 1,
-                    trees: product.children,
-                  },
-                );
+                return applyOneReactantReaction(reactionProduct, reactions, {
+                  ...options,
+                  currentDepth: options.currentDepth + 1,
+                  trees: product.children,
+                });
               });
             }
           }
           // if there is at least one product, we add the reaction to the results
           if (products.length > 0) {
-            // eslint-disable-next-line no-unused-vars
-            const { oclReaction, needToBeCharged, ...reactionWithoutOCL } =
-              reaction;
+            const { oclReaction, ...reactionWithoutOCL } = reaction;
```

Comment:
We have an eslint-error but I don't understand why. It seems to me that @targos you allowed this explicitely in https://github.com/cheminfo/eslint-config-cheminfo-typescript/commit/5c77912979c6d06aff850ffc86433114d3461914 (`ignoreRestSiblings: true`)

Reply:
TypeScript rules are only applied to TypeScript files.